src/ensembl/src/content/app/browser/drawer/drawerActions.ts

```diff
@@ -1,16 +1,61 @@
 import { createAction } from 'typesafe-actions';
 
 import { getDrawerAnalyticsObject } from 'src/analyticsHelper';
+import { ActionCreator, Action, Dispatch } from 'redux';
+import { ThunkAction } from 'redux-thunk';
+import { RootState } from 'src/store';
+import { getBrowserActiveGenomeId } from '../browserSelectors';
 
-export const changeDrawerView = createAction(
-  'drawer/change-drawer-view',
+export const changeDrawerViewForGenome = createAction(
+  'drawer/update-drawer-view',
   (resolve) => {
-    return (drawerView: string) =>
+    return (drawerView: { [genomeId: string]: string }) =>
       resolve(drawerView, getDrawerAnalyticsObject('User Interaction'));
   }
 );
 
-export const toggleDrawer = createAction('drawer/toggle-drawer', (resolve) => {
-  return (drawerOpened?: boolean) =>
-    resolve(drawerOpened, getDrawerAnalyticsObject('User Interaction'));
-});
+export const changeDrawerView: ActionCreator<
+  ThunkAction<void, any, null, Action<string>>
+> = (drawerViewForGenome: string) => (
+  dispatch: Dispatch,
+  getState: () => RootState
+) => {
+  const activeGenomeId = getBrowserActiveGenomeId(getState());
+
+  if (!activeGenomeId) {
+    return;
+  }
+
+  const drawerView = {
+    [activeGenomeId]: drawerViewForGenome
+  };
+
+  dispatch(changeDrawerViewForGenome(drawerView));
+};
+
+export const toggleDrawerForGenome = createAction(
+  'drawer/toggle-drawer',
+  (resolve) => {
+    return (drawerOpened: { [genomeId: string]: boolean }) =>
+      resolve(drawerOpened, getDrawerAnalyticsObject('User Interaction'));
```

Comment:
"User Interaction" is a useless analytics event

Reply:
I agree its name useless, but I think we'll still need to leave this to track how many times a user opens a drawer. So perhaps changing the event to a more meaningful name will resolve this?

Reply:
I have removed the analytics event for this. As discussed offline, we can add this later.